plugins/modules/ndo_mac_sec_policy.py

```diff
@@ -0,0 +1,424 @@
+                    )
+                    match.details["macsecParams"]["confOffSet"] = "offset" + str(confidentiality_offset)
+
+                if key_server_priority and match.details.get("macsecParams")["keyServerPrio"] != key_server_priority:
+                    ops.append(dict(op="replace", path="{0}/{1}/macsecParams/keyServerPrio".format(path, match.index), value=key_server_priority))
+                    match.details["macsecParams"]["keyServerPrio"] = key_server_priority
+
+            if mac_sec_keys:
+                mac_sec_keys_list = []
+                for mac_sec_key in mac_sec_keys:
+                    mac_sec_keys_list.append(
+                        dict(
+                            keyname=mac_sec_key.get("key_name"),
+                            psk=mac_sec_key.get("psk"),
+                            start=mso.verify_time_format(mac_sec_key.get("start_time")) if mac_sec_key.get("start_time") else None,
+                            end=mso.verify_time_format(mac_sec_key.get("end_time")) if mac_sec_key.get("end_time") else None,
+                        )
+                    )
+
+                ops.append(dict(op="replace", path="{0}/{1}/macsecKeys".format(path, match.index), value=mac_sec_keys_list))
+                match.details["macsecKeys"] = mac_sec_keys
+            elif mac_sec_keys == []:
+                # remove mac_sec_keys if the list is empty
+                ops.append(dict(op="remove", path="{0}/{1}/macsecKeys".format(path, match.index)))
+                match.details.pop("macsecKeys")
+
+            mso.sanitize(match.details)
+
+        else:
+            mac_sec_param_map = {}
+
+            payload = {"name": mac_sec_policy, "templateId": mso_template.template.get("templateId"), "schemaId": mso_template.template.get("schemaId")}
+            payload["adminState"] = admin_state
+            payload["type"] = interface_type
+
+            if description:
+                payload["description"] = description
+
+            if cipher_suite:
+                mac_sec_param_map["cipherSuite"] = NDO_CIPHER_SUITE_MAP.get(cipher_suite)
+            if window_size:
+                mac_sec_param_map["windowSize"] = window_size
+            if security_policy:
+                mac_sec_param_map["securityPol"] = NDO_SECURITY_POLICY_MAP.get(security_policy)
+            if sak_expiry_time:
+                mac_sec_param_map["sakExpiryTime"] = sak_expiry_time
+
+            if interface_type == "access":
+                if confidentiality_offset:
+                    mac_sec_param_map["confOffSet"] = "offset" + str(confidentiality_offset)
+                if key_server_priority:
+                    mac_sec_param_map["keyServerPrio"] = key_server_priority
+
+                payload["macsecParams"] = mac_sec_param_map
+
+            mac_sec_keys_list = []
+            if mac_sec_keys:
```

Comment:
Would it be good to fail the module if more than one mac_sec_key is included during its creation?

Reply:
Now that the API response is changed,more than one mac_sec_keys can be added without any issue, previous statements in documentation was also removed after the changes.